src/algorithms.jl

```diff
@@ -0,0 +1,78 @@
+abstract type AbstractEnclosureAlgorithm end
+
+"""
+   NaturalEnclosure <: AbstractEnclosureAlgorithm
+
+Data structure to compute the range of `f` over `X` using natural enclosure, that is simply
+evaluates `f(X)` using interval arithmetic.
```

Comment:
yeah it's tricky to give a meaningful description to those, especially since this "type-based dispatch of solvers" is so unique as paradigm that you cannot assume the average user is familiar to it (examples on how to use it to be added once you can actually use those).

What about

> Data type corresponding to the natural enclosure algorithm, which computes the range blah blah by simply blah blah

maybe data type is slightly better than algorithm (which is better than data structure anyway)
```julia
function enclose(f, X, na::NaturalEnclosure)
  # implement algorithm here (or call function with algorithm implementation
end

Reply:
hmm to me that's an implementation detail, i would call still call it `Algorithm` or `Enclosure method`